test/e2e/aks_azure_cluster_autoscaler.go

```diff
@@ -0,0 +1,83 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2022-03-01/containerservice"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+	infrav1exp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+type AKSAzureClusterAutoscalerSettingsSpecInput struct {
+	Cluster       *clusterv1.Cluster
+	WaitIntervals []interface{}
+}
+
+func AKSAzureClusterAutoscalerSettingsSpec(ctx context.Context, inputGetter func() AKSAzureClusterAutoscalerSettingsSpecInput) {
+	input := inputGetter()
+
+	settings, err := auth.GetSettingsFromEnvironment()
+	Expect(err).NotTo(HaveOccurred())
+	subscriptionID := settings.GetSubscriptionID()
+	auth, err := settings.GetAuthorizer()
+	Expect(err).NotTo(HaveOccurred())
+	mgmtClient := bootstrapClusterProxy.GetClient()
+	Expect(mgmtClient).NotTo(BeNil())
+	containerserviceClient := containerservice.NewManagedClustersClient(subscriptionID)
+	containerserviceClient.Authorizer = auth
+
+	amcp := &infrav1exp.AzureManagedControlPlane{}
+	err = mgmtClient.Get(ctx, types.NamespacedName{
+		Namespace: input.Cluster.Spec.ControlPlaneRef.Namespace,
+		Name:      input.Cluster.Spec.ControlPlaneRef.Name,
+	}, amcp)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Verify that it's not set before making any changes.
```

Comment:
Lately we've been trying to structure new tests like this to be idempotent and not to assume anything about the initial state of the cluster to help make them easier to plug in to other tests down the road.

The latest test I added in #2917 establishes a decent pattern for this I think. Roughly:
1. Save the initial state
2. Delete the existing state (which might be a no-op)
3. Create
4. Update
5. Update back to the initial state (effectively a delete when the initial state was empty)

Reply:
I made these changes. Can you take a look and let me know if it matches what you were expecting?

Reply:
Overall looks good! I made the above suggestion to model after that tags test before I understood there's a possibility we might not expose a "delete" so then it makes sense if this new test isn't a perfect 1:1 match with that one.